app/src/main/java/com/example/killerapp/MainActivity.java

```diff
@@ -0,0 +1,199 @@
+                !=PackageManager.PERMISSION_GRANTED)
+            ActivityCompat.requestPermissions(this, permissions, 0);
+    }
+
+
+    /***
+     * increase the alarm volume to maximum and start the default alarm
+     * the alarm stops when the activity is closed
+     */
+    public void startAlarm()
+    {
+        MediaPlayer mediaPlayer =MediaPlayer.create(this,
+                Settings.System.DEFAULT_RINGTONE_URI);
+        AudioManager audioManager= (AudioManager) getSystemService((Context.AUDIO_SERVICE));
+        assert audioManager != null;
+        audioManager.setStreamVolume(AudioManager.STREAM_MUSIC,audioManager.getStreamMaxVolume(AudioManager.STREAM_MUSIC),0);
+        mediaPlayer.start();
+
+    }
+
+
+    /***
+     * get the current phone's gps coordinates and write them on the "gps coordinates" editable text
+     * @param gpsCoordinatesRefreshDistance refresh the gpsCoordinate when you travel gpsCoordinateRefreshDistance meters
+     * @param gpsCoordinatesRefreshTime refresh the gpsCoordinate every gpsCoordinateRefreshTime seconds
+     */
+    public void getCoordinates(int gpsCoordinatesRefreshTime,int gpsCoordinatesRefreshDistance)
+    {
+        locationManager = (LocationManager) getSystemService(LOCATION_SERVICE);
+        locationListener= new LocationListener() {
+            @Override
+            public void onLocationChanged(Location location) {
+
+            }
+
+            @Override
+            public void onStatusChanged(String provider, int status, Bundle extras) {
+
+            }
+
+            @Override
+            public void onProviderEnabled(String provider) {
+
+            }
+
+            @Override
+            public void onProviderDisabled(String provider) {
+
+            }
+        };
+        if (((ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) +
+                ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_COARSE_LOCATION)) !=PackageManager.PERMISSION_GRANTED))
+        {
+            ActivityCompat.requestPermissions(this, new String[]{Manifest.permission.ACCESS_FINE_LOCATION},REQUEST_GPS_COARSE_LOCATION );
+            ActivityCompat.requestPermissions(this, new String[]{Manifest.permission.ACCESS_COARSE_LOCATION},REQUEST_GPS_FINE_LOCATION );
+        }
+        locationManager.requestLocationUpdates("gps", gpsCoordinatesRefreshTime, gpsCoordinatesRefreshDistance, locationListener);
+        Location location=locationManager.getLastKnownLocation(LocationManager.GPS_PROVIDER);
+        assert location != null;
+        gpsCoordinates.append("\n "+location.getLongitude()+"\n "+location.getLatitude());
+    }
+
+
+    /***
+     * when a message is received with only the key word it open a new empty activity,start an alarm and send a message ,to the phone number
+     * he received the message from ,with the device's current gps coordinates
+     * @param message Received SMSMessage class of SmsHandler library
```

Comment:
Il codice è scritto molto bene e non ho nulla da aggiungere se non una piccola nota per questa specifica sull'uso dell'inglese (per esempio "opens" alla prima riga per la terza persona singolare, oppure "it" al posto di "he" all'inizio della seconda riga, essendo il telefono un oggetto).

Reply:
Thanks for the suggestion i'll try to check a bit better my comments next time